service/history/historyBuilder.go

```diff
@@ -750,15 +765,17 @@ func (b *historyBuilder) newRequestCancelExternalWorkflowExecutionInitiatedEvent
 func (b *historyBuilder) newRequestCancelExternalWorkflowExecutionFailedEvent(workflowTaskCompletedEventID, initiatedEventID int64,
 	namespace, workflowID, runID string, cause enumspb.CancelExternalWorkflowExecutionFailedCause) *historypb.HistoryEvent {
 	event := b.msBuilder.CreateNewHistoryEvent(enumspb.EVENT_TYPE_REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION_FAILED)
-	attributes := &historypb.RequestCancelExternalWorkflowExecutionFailedEventAttributes{}
-	attributes.WorkflowTaskCompletedEventId = workflowTaskCompletedEventID
-	attributes.InitiatedEventId = initiatedEventID
-	attributes.Namespace = namespace
-	attributes.WorkflowExecution = &commonpb.WorkflowExecution{
-		WorkflowId: workflowID,
-		RunId:      runID,
+	attributes := &historypb.RequestCancelExternalWorkflowExecutionFailedEventAttributes{
+		WorkflowTaskCompletedEventId: workflowTaskCompletedEventID,
+		InitiatedEventId:             initiatedEventID,
+		Namespace:                    namespace,
+		WorkflowExecution: &commonpb.WorkflowExecution{
+			WorkflowId: workflowID,
+			RunId:      runID,
+		},
+		Cause:   cause,
+		Control: "",
```

Comment:
I guess "Control" was never used?

Reply:
Yes, I was about to remove it from everywhere but then we decided to left it. I think this particular instance is not used.